src/models/unbound_step_definition.js

```diff
@@ -0,0 +1,21 @@
+import Definition from './definition'
+import StepDefinition from './step_definition'
+import { CucumberExpression, RegularExpression } from 'cucumber-expressions'
+
+export default class UnboundStepDefinition extends Definition {
```

Comment:
Yeah I see your point. Passing the parameter type registry into the StepDefinition constructor didn't feel right to me, because it wouldn't need the parameter type registry after creating the relevant expression. Since we have this build_helpers module which already knows about cucumber-expressions, I've deleted the unnecessary class and moved the responsibility of creating the expression there. Thought I'd see if you'd want to leave it like that or keep going?